pkg/networkutils/network.go

```diff
@@ -740,13 +746,34 @@ func (r iptablesRule) String() string {
 	return fmt.Sprintf("%s/%s rule %s shouldExist %v rule %v", r.table, r.chain, r.name, r.shouldExist, r.rule)
 }
 
+// netLinkRuleDelAll deletes all matching route rules (instead of only first instance).
+func NetLinkRuleDelAll(netlink netlinkwrapper.NetLink, rule *netlink.Rule) error {
+	for {
+		if err := netlink.RuleDel(rule); err != nil {
+			if !containsNoSuchRule(err) {
+				return err
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func ContainsNoSuchRule(err error) bool {
```

Comment:
Are these introduced for backwards compatibility instead of refactoring the existing functions?

Reply:
They are moved from the deleted `cmd/routed-eni-cni-plugin/driver/utils.go` so that they can be used by IPAMD and CNI logic without duplicating code